tests/generator-Tests/Unit-Tests/CodeGeneratorTests.cs

```diff
@@ -827,6 +827,55 @@ public void DontWarnIfNestedTypeNameMatchesNamespace ()
 			Assert.False (sb.ToString ().Contains ("warning BG8403"));
 		}
 
+
+		[Test]
+		public void AvoidNREOnInvalidBaseMethod ()
+		{
+			// We copy methods from the package-private base class to the public class, however
+			// the copied method is not valid because it doesn't understand the generic argument
+			// type. The method is remove from the public class, but we need to ensure the
+			// base class method still exists and IsValid.
+			var xml = @"<api>
+			  <package name='java.lang' jni-name='java/lang'>
+			    <class abstract='false' deprecated='not deprecated' final='false' name='Object' static='false' visibility='public' jni-signature='Ljava/lang/Object;' />
+			  </package>
+
+			  <package name='android.view' jni-name='android/view'>
+			    <class abstract='false' deprecated='not deprecated' extends='java.lang.Object' final='false' name='View' static='false' visibility='public' jni-signature='Landroid/view/View;' />
+			  </package>
+
+			  <package name='com.google.android.material.behavior' jni-name='com/google/android/material/behavior'>
+			    <class abstract='false' deprecated='not deprecated' extends='java.lang.Object' extends-generic-aware='java.lang.Object' jni-extends='Ljava/lang/Object;' final='false' name='ViewOffsetBehavior' static='false' visibility='' jni-signature='Lcom/google/android/material/appbar/ViewOffsetBehavior;'>
+			      <typeParameters>
+				<typeParameter name='V' classBound='android.view.View' jni-classBound='Landroid/view/View;'>
+				  <genericConstraints>
+				    <genericConstraint type='android.view.View' />
+				  </genericConstraints>
+				</typeParameter>
+			      </typeParameters>
+			      <method abstract='false' deprecated='not deprecated' final='false' name='layoutChild' jni-signature='(Landroid/view/View;)V' bridge='false' native='false' return='void' jni-return='V' static='false' synchronized='false' synthetic='false' visibility='protected'>
+				<parameter name='child' type='V' jni-type='TV;' not-null='true' />
+			      </method>
+			    </class>
+    
+			    <class abstract='false' deprecated='not deprecated' extends='com.google.android.material.behavior.ViewOffsetBehavior' extends-generic-aware='com.google.android.material.behavior.ViewOffsetBehavior&lt;android.view.View&gt;' jni-extends='Lcom/google/android/material/appbar/ViewOffsetBehavior;' final='false' name='Behavior' static='false' visibility='public' jni-signature='Lcom/google/android/material/appbar/Behavior;'>
+			    </class>
+			  </package>
+			</api>";
+
+			var gens = ParseApiDefinition (xml);
+
+			var public_class = gens.Single (g => g.Name == "Behavior");
+			var base_class = gens.Single (g => g.Name == "ViewOffsetBehavior");
+
+			// Method got removed
+			Assert.AreEqual (0, public_class.Methods.Count);
```

Comment:
Why is this asserting that the method got removed?

OK, this test is asserting that we no longer NRE, that the BG8800 warning is still emitted (implicitly), that `public_class` has no methods, and `base_class.Methods` is still sane.

But… shouldn't the "final" fix be a situation in which:

 1. `public_class.Methods.Count` is *also* 1, *and*
 2. `public_class.Methods[0].IsValid` is true, *and*
 3. `public_class.Methods[0].Parameters[0].JniType` is `java/lang/Object`?

Upon review, it feels like this PR fixes the NRE, but doesn't resolve the underlying scenario allowing publicly accessible members from package-private types to be bound in the derived type.

Is that possible?

Reply:
Correct, this PR only fixes the NRE.  It does not try to fix the underlying issue of correctly supporting generics when copying methods from a `package-private` class.  I do not believe that is a topic we want to tackle any time soon.